### ISSUE TITLE
[BUG] Card Refresh jQuery API not working

### ISSUE THREAD

**Issue by emregulerr** [ColorlibHQ/AdminLTE#2827]
**Describe the bug**
Card Refresh is working through data api but the jQuery way isn't working...

**Environment :**
- AdminLTE Version: [v3.0.5]
- Operating System:  [macOS Catalina]
- Browser (Version): [Safari (Latest)]

**Additional context**
ERROR: 
.refreshBox is not a function.

Please also provide a way to refresh card by setInterval (usage of load method?)

**Comment by emregulerr** [comment]
I solved the problem about reloading via jQuery.
There is a mistake in the sample code written in the documentation.
After initializing with the Data API, I can reload it with jQuery.

The code should be: 
`$('#my-card-widget').CardRefresh('toggle')`
 instead of:
`$('#my-card-widget').Widget('toggle')`

But I still can't initialize it via jQuery.

**Comment by REJack** [comment]
It would be helpful if you can post your JS Code to init/load the CardRefresh JS.

**Comment by emregulerr** [comment]
I've tried: 
`$("#my-card").refreshBox({source: 'my-ajax-url'});`
and 
`$("#my-card").cardRefresh({source: 'my-ajax-url'});`
and 
`$("#my-card").boxRefresh({source: 'my-ajax-url'});`
and 
`$("#my-card").refreshCard({source: 'my-ajax-url'});`
and 
`$("#my-card").CardRefresh({source: 'my-ajax-url'});`

but none of them worked. All returned an error that says it's not a function except CardRefresh. But CardRefresh returned an error like; 'sourceURL is empty, please provide source option'. 
I also tried:
`$("#my-card").CardRefresh('load',{source: 'my-ajax-url'});`
and
`$("#my-card").CardRefresh($(this),{source: 'my-ajax-url'});`

**Comment by XhmikosR** [comment]
The only public API is `load()`. The docs are outdated I think.

**Comment by emregulerr** [comment]
@XhmikosR how can I use it?

**Comment by XhmikosR** [comment]
Not sure at this point, something is clearly wrong either in docs or both in docs and code.

**Comment by REJack** [comment]
I see what you mean, the code doesn't work like the docs explains it 🤣.

It looks like any of the plugins doesn't work correcty via jQuery API. 
I need to redesign the _jQueryInterface to get this working like it should for now this looks like the right approach:
```js
let data = $(this).data(DATA_KEY)

if (!data) {
  data = $(this).data()
}

const _options = $.extend({}, Default, typeof config === 'object' ? config : data)
const plugin = new CardRefresh($(this), _options)

$(this).data(DATA_KEY, typeof config === 'object' ? config : data)

if (typeof config === 'string' && config.match(/load/)) {
  plugin[config]()
} else {
  plugin._init($(this))
}
```

This works with data attributes and jQuery API.

@XhmikosR what you think about my reworked _jQueryInterface?

**Comment by emregulerr** [comment]
Thanks, now it works like this:

`$("#my-card").CardRefresh({source: 'my-ajax-url'});`

**Comment by XhmikosR** [comment]
@REJack I haven't tested the code, but some notes:

1. you need to verify all plugins work and do the same where needed
2. might be worth adding a new private function to get the options if you need it more than once in a plugin
3. check how we do it upstream to get more ideas
4. that's why the codebase needs tests...

**Comment by taufardh** [comment]
@XhmikosR so this bug is fixed in v3.1.0?

**Comment by danny007in** [comment]
> 
> 
> @XhmikosR so this bug is fixed in v3.1.0?

Probably Yes,
Once check it

**Comment by Elecktro1337** [comment]
> I see what you mean, the code doesn't work like the docs explains it 🤣.
> 
> It looks like any of the plugins doesn't work correcty via jQuery API. I need to redesign the _jQueryInterface to get this working like it should for now this looks like the right approach:
> 
> ```js
> let data = $(this).data(DATA_KEY)
> 
> if (!data) {
>   data = $(this).data()
> }
> 
> const _options = $.extend({}, Default, typeof config === 'object' ? config : data)
> const plugin = new CardRefresh($(this), _options)
> 
> $(this).data(DATA_KEY, typeof config === 'object' ? config : data)
> 
> if (typeof config === 'string' && config.match(/load/)) {
>   plugin[config]()
> } else {
>   plugin._init($(this))
> }
> ```
> 
> This works with data attributes and jQuery API.
> 
> @XhmikosR what you think about my reworked _jQueryInterface?

Srry, where i must paste this?
I have 3.2 version with sub porblem...
